ast_tools/passes/ssa.py

```diff
@@ -514,14 +549,22 @@ def leave_Name(self,
             # Names in Load context should not be added to the name table
             # as it makes them seem like they have been modified.
             try:
-                return cst.Name(self.name_table[name])
+                new_name = self.name_table[name]
             except KeyError:
                 if self.strict:
                     raise SyntaxError(f'Cannot prove name `{name}` is defined')
-                else:
-                    return cst.Name(name)
+                new_name = name
         else:
-            return cst.Name(self._make_name(name))
+            new_name = self._make_name(name)
+        line = self.get_metadata(PositionProvider, original_node).start.line
+        if not self.symbol_table_skip:
+            # TODO: What should we use if we have something like x = .... + x?
+            # Should it be the original (read) value or the new (write) value?
+            # Based on the traversal order, right now it will get the assign
+            # target value
+            offset_line = line - self.symbol_table_offset
```

Comment:
This seems overly convoluted.   Should just resolve PositionProvider on the original tree (before any edits)

Reply:
Also not sure if this accurate because the normalizers (ElifToElse) will probably change line numbering

Reply:
Hmm, resolving on the original tree isn't quite working as I would expect, I have to look more closely but the original name nodes aren't being found in the dictionary.   The pattern looks similar to the expression context resolve so I'm not sure why the node lookup isn't working, maybe there's a copy occurring somewhere between the start and ssa transformer?